### DISCUSSION TITLE
Updating 3rd party app

### DISCUSSION THREAD

**Discussion by eimantas** [sparkle-project/Sparkle#2227] [Ideas]
I haven't found (or rather didn't know which keywords to use for) a discussion on how I could approach a 3rd party application update.

Scenario:
1. there exists an app that isn't distributed through MAS and the only way to get updates is to read about them in twitter feed, go to the site and fetch latest dmg;
2. due to licensing and culture the application can't connect to the web w/o user's explicit consent and interactions
3. app is also available on Windows and Linux thus requires features that would serve all platforms;
4. there are other 3rd party updaters for the app on Windows and Linux platforms

The question (finally) is: is it feasible to make my own application with embedded Sparkle framework so that I would be able update THAT application, not my own. I understand that I would need to create my own appcast with links and signatures for that app. But I wonder if Sparkle would facilitate update and copying the code to update the binary.

**Comment by zorgiepoo** [comment]
Check discussions like #2141 and [sparkle-cli](https://sparkle-project.org/documentation/sparkle-cli/)

**Reply by eimantas** [reply]
This is not the case I am writing about. The case I am writing about is whether I can update an application without IT having Sparkle integrated, but me providing an appcast with links to its downloads and my app doing the checks.

**Comment by zorgiepoo** [comment]
So the question is can you use Sparkle to update an app that doesn't integrate Sparkle framework itself?

I think it's possible (note I haven't tried) if the apps are code-signed (probably with an Apple developer ID certificate) and you serve the appcast feed and downloads via https. Since you don't have control of the app in question, it probably can't(?) specify the public EdDSA key in the app's Info.plist so you may not be able to use EdDSA signing on the update archive (so then https + apple code signing is necessary).

You probably need to implement [`-feedURLStringForUpdater:`](https://sparkle-project.org/documentation/api-reference/Protocols/SPUUpdaterDelegate.html#/c:objc(pl)SPUUpdaterDelegate(im)feedURLStringForUpdater:) delegate method to pass the feed URL. If you use sparkle-cli, that is through the `--feed-url` argument.
